measurement/plugins/netflix_fast/measurements.py

```diff
@@ -0,0 +1,334 @@
+            results.append(self._get_url_result(thread_result))
+        return results
+
+    def _get_fast_result(self):
+        s = requests.Session()
+        try:
+            resp = self._get_response(s)
+        except ConnectionError as e:
+            return self._get_netflix_error("netflix-response", traceback=str(e))
+
+        try:
+            script = re.search(r'<script src="(.*?)">', resp.text).group(1)
+        except AttributeError:
+            return self._get_netflix_error("netflix-script-regex", traceback=resp.text)
+
+        try:
+            script_resp = s.get("https://fast.com{script}".format(script=script))
+        except ConnectionError as e:
+            return self._get_netflix_error("netflix-script-response", traceback=str(e))
+
+        try:
+            token = re.search(r'token:"(.*?)"', script_resp.text).group(1)
+        except AttributeError:
+            return self._get_netflix_error(
+                "netflix-token-regex", traceback=script_resp.text
+            )
+
+        try:
+            self._query_api(s, token)
+        except ConnectionError as e:
+            return self._get_netflix_error("netflix-api-response", traceback=str(e))
+        except json.decoder.JSONDecodeError as e:
+            return self._get_netflix_error("netflix-api-json", traceback=str(e))
+        except TypeError as e:
+            return self._get_netflix_error("netflix-api-parse", traceback=str(e))
+        except KeyError as e:
+            return self._get_netflix_error("netflix-api-parse", traceback=str(e))
+
+        try:
+            conns = [
+                self._get_connection(target["url"]) for target in self.thread_results
+            ]
+        except ConnectionError as e:
+            return self._get_netflix_error("netflix-connection", traceback=str(e))
+
+        fast_data = self._manage_threads(conns)
+
+        return NetflixFastMeasurementResult(
+            id=self.id,
+            download_rate=float(fast_data["speed_mbps"]),
+            download_rate_unit=NetworkUnit("Mbit/s"),
+            download_size=float(fast_data["total"]),
+            download_size_unit=StorageUnit("B"),
+            asn=self.client_data["asn"],
+            ip=self.client_data["ip"],
+            isp=self.client_data["isp"],
+            city=self.client_data["location"]["city"],
+            country=self.client_data["location"]["country"],
+            urlcount=self.urlcount,
+            reason_terminated=fast_data["reason_terminated"],
+            errors=[],
+        )
+
+    def _manage_threads(self, conns):
+        # Create worker threads
+        threads = [None] * len(self.thread_results)
+        for i in range(len(self.thread_results)):
+            threads[i] = Thread(
+                target=self._threaded_download,
+                args=(conns[i], self.thread_results[i], time.time()),
+            )
+            threads[i].daemon = True
+            threads[i].start()
+
+        start_time = time.time()
```

Comment:
I think a note here stating that the time wont be perfect would be useful. As we are starting the thread downloading before we take the start time there will be a slight discrepancy.